sound/soc/sof/ipc4-topology.c

```diff
@@ -1116,6 +1126,30 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 
 		pipe_widget = swidget->pipe_widget;
 		pipeline = pipe_widget->private;
+
+		if (pipeline->use_chain_dma) {
+			u32 host_dma_id;
+			u32 fifo_size;
+
+			host_dma_id = platform_params->stream_tag - 1;
+			pipeline->msg.primary &= ~SOF_IPC4_GLB_CHAIN_DMA_HOST_ID_MASK;
+			pipeline->msg.primary |= SOF_IPC4_GLB_CHAIN_DMA_HOST_ID(host_dma_id);
+
+			if (params_width(fe_params) > 16)
```

Comment:
a comment here saying "Set SCS bit for all formats other than S16_LE"

Reply:
Then probably better to change the condition to exactly that too.